stock_location_storage_type_strategy/models/stock_location.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, models, fields
+from odoo.osv.expression import AND, OR
+
+
+class StockLocation(models.Model):
+
+    _inherit = 'stock.location'
+
+    pack_storage_strategy = fields.Selection(
+        selection=[('none', 'None'),
+                   ('ordered_locations', 'Ordered Children Locations')],
+        required=True,
+        default='none',
+        string='Packs storage strategy',
+        help='TODO',
+    )
+    display_pack_storage_strategy = fields.Boolean(
+        compute='_compute_display_pack_storage_strategy'
+    )
+
+    @api.depends(
+        'child_ids', 'location_id', 'location_id.pack_storage_strategy',
+        'location_id.display_pack_storage_strategy'
+    )
+    def _compute_display_pack_storage_strategy(self):
+        for location in self:
+            # Do not display on locations without children
+            if not location.child_ids:
+                location.display_pack_storage_strategy = False
+                continue
+            current_location = location.location_id
+            display = True
+            while current_location:
+                # If a parent has pack_storage_strategy set, we don't display
+                # it on its children
+                if current_location.pack_storage_strategy != 'none':
+                    display = False
+                    break
+                current_location = current_location.location_id
+            location.display_pack_storage_strategy = display
+
+    def get_putaway_strategy(self, product):
+        putaway_location = super().get_putaway_strategy(product)
+        quant = self.env.context.get('storage_quant')
```

Comment:
```suggestion
        quant = self.env.context.get('storage_quant')
        return self.get_storage_type_strategy(product, putaway_location, quant)
        
    def get_storage_type_strategy(self, product, putaway_location, quant=False)
```

Can we do something like this? It helps for inheritance, if we have to insert something between the put-away strategy and the storage strategy.